### PR TITLE
Updated to refactored version

### PR DESCRIPTION
**Pull request by specialunderwear**

### CODE REVIEW

File: oscar_elasticsearch/search/search_handlers.py [hunk 1]
```diff
@@ -143,26 +153,46 @@ def __init__(self, query=None):
...
+            request_data,
+            full_path,
+            facets=facets,
+            order_by_relevance=order_by_relevance,
+        )
+
+    def get_query(self):
+        filters, query = super().get_query()
+        if settings.FILTER_AVAILABLE:
+            filters["is_available"] = True
```
Comment (viggo-devries) [inline oscar_elasticsearch/search/search_handlers.py:194]: filters["is_available"] = settings.FILTER_AVAILABLE ?

File: oscar_elasticsearch/search/search_handlers.py [hunk 2]
```diff
@@ -32,16 +32,17 @@ def items(self):
 class SearchHandler(object):
     model = None
     form_class = BaseSearchForm
+    suggestion_field_name = "title"
```
Comment (viggo-devries) [inline oscar_elasticsearch/search/search_handlers.py:35]: Moet dit geen settings zijn?

**Review by viggo-devries — CHANGES_REQUESTED** [review]

File: oscar_elasticsearch/search/search_handlers.py (hunk 1 above)
Comment (specialunderwear) [reply oscar_elasticsearch/search/search_handlers.py:194]: No that would show al the products that are not availble lmao

Comment (viggo-devries) [reply oscar_elasticsearch/search/search_handlers.py:194]: filters["is_available"] = not settings.FILTER_AVAILABLE I mean :)

File: oscar_elasticsearch/search/search_handlers.py (hunk 2 above)
Comment (specialunderwear) [reply oscar_elasticsearch/search/search_handlers.py:35]: no if you want to change it you can override it, it is very unlikely that you want something else for this setting

File: README.md [hunk 3]
```diff
@@ -12,77 +12,73 @@ INSTALLED_APPS = [
 ] + get_core_apps([
     "oscar_elasticsearch.search", # replace standard oscar app
 ]) + [
+    "widget_tweaks",
     "wagtail.core",
-    "wagtail.contrib.search_promotions", # allow search result display and such
     "wagtail.search",
```
Comment (sasha0) [inline README.md:17]: Will we remove other wagtail apps as well?

File: README.md [hunk 4]
```diff
@@ -12,77 +12,73 @@ INSTALLED_APPS = [
...
 ]
 ```
 
 Settings required to replace the search app.
 
 ```Python
-OSCAR_ELASTICSEARCH_QUERY_PAGE_SIZE = 100
 OSCAR_PRODUCT_SEARCH_HANDLER = "oscar_elasticsearch.search.search_handlers.ProductSearchHandler"
-OSCAR_SEARCH = {
```
Comment (sasha0) [inline README.md:26]: Why this setting was removed?

File: oscar_elasticsearch/search/search_handlers.py (hunk 1 above)
Comment (sasha0) [reply oscar_elasticsearch/search/search_handlers.py:194]: Yeah, it would be shorter.

**Review by sasha0 — COMMENTED** [review]
Looks good in general. Although not sure how to get Wagtail fully extracted yet.

File: README.md (hunk 3 above)
Comment (specialunderwear) [reply README.md:17]: Hmm maybe we can remove wagtail.core, I think the wagtail.search has to stay because of the query stats model

Comment (specialunderwear) [reply README.md:17]: I tried, it's not possible in the current situation

File: README.md (hunk 4 above)
Comment (specialunderwear) [reply README.md:26]: The settings are nolonger grouped but separate values, with sane defaults.

**Pull request merged**